src/frequenz/sdk/model/_handler.py

```diff
@@ -0,0 +1,448 @@
+                )
+            return self._period_to_num[index]
+
+        interval_period = f"{index.hour:02d}{index.minute:02d}"
+        sorted_keys = sorted(self._period_to_num.keys())
+        position = bisect.bisect(sorted_keys, interval_period)
+        number = sorted(self._period_to_num.values())[position - 1]
+        return number
+
+
+@dataclass(frozen=True)
+class ModelHandlerConfig(Generic[T]):
+    """Configuration for the model handler."""
+
+    model_cls: Type[T]
+    """The model class to load, store, update, and retrieve."""
+
+    model_dir_prefix: str
+    """The directory to the model definition or models' parent directory.
+    
+    This attribute represents the directory path where the model's definition
+    is stored in the case of a single model. In the case of multiple models,
+    it refers to the parent directory containing the subdirectories for each
+    model.
+
+    For example, if each model has its own subdirectory (e.g., "mon", "tue", etc.) 
+    and are stored in the directories "/models/consumption/day/mon", 
+    "/models/consumption/day/tue", etc., the `model_dir_prefix` needs to be
+    set to "/models/consumption/day" to specify the parent directory.
+
+    Note that `model_dir_prefix` can be either a relative path, relative to
+    the current working directory, or an absolute path specifying the full
+    directory path.
+    """
+
+    model_file_name: str = "model.pkl"
+    """The name of the file used to store the model."""
+
+
+class ModelHandler(Generic[T]):
+    """Interface for loading, updating, and monitoring machine learning models."""
+
+    def __init__(
+        self,
+        config: ModelHandlerConfig[T],
+        policy: ModelKeyPolicy,
+    ) -> None:
+        """Initialize the model handler to handle machine learning models.
+
+        Args:
+            config: the configuration for the model handler.
+            policy: the model key policy used to retrieve models.
+
+        Note:
+            The suggested directory tree for model(s) definition is as follows:
+
+            models/
+                {consumption,production}/
+                    single/
+                        model.pkl
+                    day/
+                        [mon,tue,wed,thu,fri,sat,sun]/
+                            model.pkl
+                    interval/
+                        [0000,0015,0030,0045,0100,...,2345]/
+                            model.pkl
+
+            In this directory tree:
+            - `models/{consumption,production}/{single,day,interval}` should be
+              set as the `model_dir_prefix` in the model handler configuration.
+            - The subdirectories [mon,tue,wed,thu,fri,sat,sun] or
+              [0000,0015,0030,0045,0100,...,2345] should match the model periods
+              defined in the respective model key policy.
+        """
+        self._model_cls: Type[T] = config.model_cls
+        self._policy: ModelKeyPolicy = policy
```

Comment:
I would name this `_key_policy`, as you might want to have other policies in the future.